graylog2-web-interface/src/views/stores/ViewStore.ts

```diff
@@ -157,7 +157,7 @@ export const ViewStore: ViewStoreType = singletonStore(
       /* Select selected query (activeQuery) or first query in view (for now).
          Selected query might become a property on the view later. */
       const queries = view.state.keySeq().toList();
-      const firstQueryId = queries.first();
+      const firstQueryId = view?.search?.queries?.first()?.id ?? queries.first();
       const selectedQuery = this.activeQuery && queries.find((id) => (id === this.activeQuery)) ? this.activeQuery : firstQueryId;
 
       this.selectQuery(selectedQuery);
```

Comment:
Do you think we should also ensure that `views.state` has the same order like `view.search.queries` on the backend side? For example to avoid similar problems in the future.

Reply:
Two thoughts:

  - In the future we should not rely on the order of the queries and store the tab order explicitly in the view.
  - For now, we should have only a single point where the order is derived from and that is the `queries` part of the `Search`.